federationapi/routing/routing.go

```diff
@@ -678,6 +679,53 @@ func MakeFedAPI(
 	return httputil.MakeExternalAPI(metricsName, h)
 }
 
+// MakeFedAPI makes an http.Handler that checks matrix federation authentication.
+func MakeFedAPIHTML(
```

Comment:
It ain't HTML though?

Reply:
While true, we also have `MakeHTMLAPI`, with basically the same function signature. That's why I named it that way, could be `MakeFedHTMLAPI`, though.